app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php

```diff
@@ -1644,7 +1644,7 @@ public function addAttributeToSort($attribute, $dir = self::SORT_ORDER_ASC)
                 return $this;
             }
             if ($this->isEnabledFlat()) {
-                $this->getSelect()->order("cat_index_position {$dir}");
+                $this->getSelect()->order("cat_index.position {$dir}");
```

Comment:
In which case we should surely take this out to prevent any further unknown column errors? I know i've seen a very similar error in Magento 1.x which is fixed through the applying the above.